### DISCUSSION TITLE
Game/Battlesnake Info pages - Display "version" tag from /info

### DISCUSSION THREAD

**Discussion by BoldBigflank** [BattlesnakeOfficial/feedback#175] [Feature Requests]
I haven't completely thought through this, and I'm open to discussion, but it would be nice to save just a little more information to a snake.

When a game is played from the battlesnake engine, it saves a copy of the snake's info meaning when you go back it has the old color/name of the snake. Rather than update the color or customizations of the snake to distinguish versions of snakes, it would be nice to have an extra field to let the developer know what version of the snake ran the game.

I propose that the /info method include a `version` field that followed SEMVER conventions. Then in the board repo, have it included as alt-text or something. In the snake's Recent Matches, include in subscript the version passed to the game engine.

I believe that the battlesnake server is currently being a little lazy and only updating the /info data every 24 hours or something, so this might also need to be updated to happen at the beginning of every game.

**Comment by BoldBigflank** [comment]
A "nice to have" extension of this feature would be a breakdown of game stats grouped by version, so all the hard work would be done and we could get information like
```
v1.1.0
291 games played
108 wins
183 losses 
v1.0.0
1000 games played
210 wins
790 losses 
```

**Comment by bvanvugt** [comment]
Huh, this is neat. Might be easy too, do you mean the version tag available here?
https://docs.battlesnake.com/references/api#get

**Comment by BoldBigflank** [comment]
Ah, I didn't realize that existed! I was using what I had seen from a starter project which didn't define that.

So I guess what I need is an easy way to see that in the views that I look to track snake progress. Places like the snake info page and the play.battlesnake.game/g/{gameId} page.

**Reply by BoldBigflank** [reply]
I'm trying it out, and it looks like the `version` tag just adds it as a tag to your snake's list of tags. I'd suggest removing it from that spot, since this is a place that people can write whatever (unless it is validating SEMVER, I haven't looked yet), plus I'd prefer it to not be a public thing, but for a developer to have extra information about their own snakes.

**Reply by bvanvugt** [reply]
Yeah, I buy this. Unsure about including it in board/api calls though, for the same validation reason. Hmm. 

For now, we'll remove the tag showing version - pretty sure we can both agree that's not terribly useful :)

**Reply by bcambl** [reply]
Sad to see the version tag go away as I frequently used it as a quick way to verify my latest version is deployed/live via the site.  
That said.. it is pretty easy to curl my snake server `/`.  
FWIW I do not use semver for my snake version. I used a truncated commit hash as my version is set via my deployment pipeline.

**Reply by BoldBigflank** [reply]
Ah, someone does use it as it exists! I feel like it will work even better for your use case once the design is ironed out.

It just needs to be visible only to the owner. Just like how tournament lists color the row of the owner's snake, the version number could be put on past games only when you are the owner of the snake.

**Comment by mawif** [comment]
I like the idea of having this saved per game; I'd prefer version be free-form, so folks can use semver if they want, but things like git hashes remain valid.

Rather than having the server poll /info more often, the server could just accept the same info payload in response to the /start call. If returning anything in response to /start remains optional, nothing breaks on the snakes side. (Obviously I have no idea if that's a trivial thing to plumb in on the server side, or would require massive refactoring efforts.)
